src/energyhub.py

```diff
@@ -96,13 +98,12 @@ def _perform_preprocessing_checks(self):
                             "SU_time",
```

Comment:
we should add here: 

1. If averaging is used, dynamics cannot be used and RR has to be -1 for all techs (RR is outside of dynamics)
2. If clustering method 1 is used: technologies cannot have a RR of >-1, min_uptime or min_downtime of >-1, SUSD duration of >-1, SUload or SDload of >-1 or performance function type=4.
3. If clustering method 2 is used, technologies that are NOT modeled with full res cannot have a min_uptime or min_downtime of >-1, SUSD duration of >-1, SUload or SDload of >-1 or performance function type=4.

What I will add after merging everything (which should be very easy if it works in one try ;)) is some lines in technology.py that say if clustering method 2 is used and you have a technology at reduced resolution, but with a RR, var_input is at full resolution and not aux.

Reply:
Could you add this?